astropy/coordinates/baseframe.py

```diff
@@ -691,10 +695,26 @@ def isscalar(self):
         return self.has_data and self.data.isscalar
 
     @classmethod
-    def get_frame_attr_names(cls):
+    def get_frame_attr_defaults(cls):
+        """Return a dict with the defaults for each frame attribute"""
         return {name: getattr(cls, name)
                 for name in cls.frame_attributes}
 
+    @deprecated(
+        "5.2",
+        alternative="get_frame_attr_defaults",
+        message=(
+            "The {func}() {obj_type} is deprecated and may be removed in a future"
```

Comment:
Just curious: Does f-string actually work on a deprecation warning message?

Reply:
`message` is not an f-string. Field replacement is done by the `str.format()` method: https://github.com/astropy/astropy/blob/75c5e83d6344ca5aacb66794e79f3c55091c91ca/astropy/utils/decorators.py#L188-L193
See also the docstring: https://github.com/astropy/astropy/blob/75c5e83d6344ca5aacb66794e79f3c55091c91ca/astropy/utils/decorators.py#L36-L42